lib/puppet/functions/azure_key_vault/secret.rb

```diff
@@ -4,33 +4,58 @@
 Puppet::Functions.create_function(:'azure_key_vault::secret', Puppet::Functions::InternalFunction) do
   # @param vault_name Name of the vault in your Azure subcription.
   # @param secret_name Name of the secret to be retrieved.
-  # @param api_versions_hash A Hash of the exact versions of the metadata_api_version and vault_api_version to use.
+  # @param api_endpoint_hash A Hash with API endpoint and authentication information
   # @param secret_version The version of the secret you want to retrieve.  This parameter is optional and if not passed the default behavior is to retrieve the latest version.
   # @return [Sensitive[String]] Returns the secret as a String wrapped with the Sensitive data type.
   dispatch :secret do
     cache_param
     required_param 'String', :vault_name
     required_param 'String', :secret_name
-    required_param 'Hash',   :api_versions_hash
+    param 'Struct[{
```

Comment:
Enforcing the type clarifies what is expected for this parameter, but there is a chance it breaks the code of current users of the module.